main.tf

```diff
@@ -231,6 +231,21 @@ resource "google_compute_address" "static_nginx_ip" {
   }
 }
 
+resource "helm_release" "nginx_controller" {
+   # see https://kubernetes.github.io/ingress-nginx/deploy/#using-helm
+  count            = var.nginx_controller.enabled ? 1 : 0
+  name             = "nginx-ingress"
+  namespace        = "nginx-ingress"
+  create_namespace = true
+  repository       = "https://kubernetes.github.io/ingress-nginx"
+  chart            = "ingress-nginx"
+  set_sensitive {
+    name  = "controller.service.loadBalancerIP"
+    value = google_compute_address.static_nginx_ip[var.nginx_controller.ip_name].address
```

Comment:
Based on that, you should make var.nginx_controller.ip_name mandatory and you should send it always, otherwise you will face a problem.

Reply:
Then it is a valid error. 
There is already a `depends_on` check we are doing to make sure that the resource `google_compute_address.static_nginx_ip` exists.

Reply:
I mean If we send the variable something like the following nginx_controller = {enabled = true}? would it work?

Reply:
In the scenario you just mentioned, you are violating the expected object definition which should contain two like this `nginx_controller           = { enabled = true, ip_name = "auth" }`, so terraform plan/apply will not execute and will show a **Error: Invalid value for module argument**. 
That is also the reason for setting the default values in the [variables.tf](https://github.com/airasia/terraform-google-gke_cluster/pull/18/files/443802b4f5827afea61e6c805e4f7034bd1f7930#diff-05b5a57c136b6ff596500bcbfdcff145ef6cddea2a0e86d184d9daa9a65a288e), without it the same error will be shown and will affect backward compatibility.